src/wp-includes/post.php

```diff
@@ -6481,6 +6493,22 @@ function wp_delete_attachment_files( $post_id, $meta, $backup_sizes, $file ) {
 		}
 	}
 
+	// Check for alternate full size mime types in the root sources array to delete.
+	if ( isset( $meta['sources'] ) && is_array( $meta['sources'] ) ) {
+		$sources = $meta['sources'];
+		array_shift( $sources );
+		foreach ( $sources as $mime => $properties ) {
+			if ( ! is_array( $properties ) || empty( $properties['file'] ) ) {
+				continue;
+			}
+
+			$intermediate_file = str_replace( wp_basename( $file ), $properties['file'], $file );
+			if ( ! wp_delete_file_from_directory( $intermediate_file, $intermediate_dir ) ) {
+				$deleted = false;
+			}
+		}
+	}
+
 	if ( is_array( $backup_sizes ) ) {
```

Comment:
In the `$backup_sizes` loop below, we also need to check for any sources and delete those files if set (i.e. iterate through `$size['sources']` if it is set. The sources are automatically backed up  even with the code here since they are within the `sizes`, so we need to cater for that and delete those additional size sources.

Reply:
This is corrected.